src/Integration.Vsix/SonarLintTracker/IssuesSnapshot.cs

```diff
@@ -0,0 +1,140 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Shell.TableControl;
+using Microsoft.VisualStudio.Shell.TableManager;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+namespace SonarLint.VisualStudio.Integration.Vsix
+{
+    internal class IssuesSnapshot : WpfTableEntriesSnapshotBase
+    {
+        private readonly string filePath;
+        private readonly int versionNumber;
+
+        private readonly IList<IssueMarker> issueMarkers;
+        private readonly IReadOnlyCollection<IssueMarker> readonlyIssueMarkers;
+        public IEnumerable<IssueMarker> IssueMarkers => readonlyIssueMarkers;
+
+        public IssuesSnapshot NextSnapshot;
+
+        internal IssuesSnapshot(string filePath, int versionNumber, IList<IssueMarker> issueMarkers)
+        {
+            this.filePath = filePath;
+            this.versionNumber = versionNumber;
+
+            this.issueMarkers = issueMarkers;
+            this.readonlyIssueMarkers = new ReadOnlyCollection<IssueMarker>(issueMarkers);
+        }
+
+        public override int Count => this.issueMarkers.Count;
+
+        public override int VersionNumber => this.versionNumber;
+
+        public override bool TryGetValue(int index, string columnName, out object content)
+        {
+            if (index < 0 || this.issueMarkers.Count <= index)
+            {
+                content = null;
+                return false;
+            }
+
+            // TODO use correct issue type, such as bug, vulnerability, code smell
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png 'Severity: INFO') Complete the task associated to this 'TODO' comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=csharpsquid%3AS1135)